fix(linter): false positive in `unicorn/escape-case`

shulaoda opened the pull request:
closes #9583, #9585

Related to #9568

This is a regression issue caused by my refactoring. I have fixed it and added corresponding test cases.

camchenry commented:
### Merge activity

* **Mar 9, 11:06 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

shulaoda committed:
fix(linter): false positive in `unicorn/escape-case` (#9638)

closes #9583, #9585

Related to #9568

This is a regression issue caused by my refactoring. I have fixed it and added corresponding test cases.
